ui/shared/status/core/StatusBadge.qml

```diff
@@ -0,0 +1,40 @@
+import QtQuick 2.13
+import "../../../imports"
+
+Rectangle {
+    id: statusBadge
+
+    property int value
+
+    implicitHeight: statusBadge.value > 0 ? 18 + statusBadge.border.width : 10 + statusBadge.border.width
+    implicitWidth: {
+        if (statusBadge.value > 99) {
+            return 28 + statusBadge.border.width
+        }
+        if (statusBadge.value > 9) {
+            return 26 + statusBadge.border.width
+        }
+        if (statusBadge.value > 0) {
+            return 18 + statusBadge.border.width
+        }
+        return 10 + statusBadge.border.width
+    }
+    radius: height / 2
+    color: Style.current.blue
+    Text {
+        id: value
+        font.pixelSize: statusBadge.value > 99 ? 10 : 12
+        color: Style.current.white
+        anchors.horizontalCenter: parent.horizontalCenter
+        anchors.verticalCenter: parent.verticalCenter
+        text: {
+            if (statusBadge.value > 99) {
+                return "99+";
+            }
+            if (statusBadge.value === 0) {
+                return "";
```

Comment:
I think you can remove this line if instead you add `visible: statusBadge.value > 0` to `Text`.